src/libAtomVM/nifs.c

```diff
@@ -4373,6 +4379,36 @@ static term nif_lists_reverse(Context *ctx, int argc, term argv[])
     return result;
 }
 
+static term nif_maps_from_keys(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+
+    term keys_tuple = nif_erlang_list_to_tuple_1(ctx, 1, argv);
+    if (UNLIKELY(term_is_invalid_term(keys_tuple))) {
+        return keys_tuple;
+    }
+
+    int map_size = term_get_tuple_arity(keys_tuple);
+
+    // argv[1] is the value
+    term roots[2] = { argv[1], keys_tuple };
```

Comment:
I would have used argv[0] (`argv[0] = keys_tuple;`), but on this other hand, this is a little hackish and could be confusing for a minimal C stack benefit.

Reply:
This function has been changed and the code looks different now.